src/data/list/basic.lean

```diff
@@ -798,6 +798,15 @@ begin
   { simp }
 end
 
+@[simp]
+lemma tail_nth_le (l : list α) (i) (h : i < l.tail.length) :
+  l.tail.nth_le i h = l.nth_le (i+1) (by { rw l.length_tail at h, exact lt_sub_iff_right.mp h, }) :=
+begin
+  induction l with a l ih,
+  { cases h, },
+  { simp, refl, }
+end
```

Comment:
```suggestion
lemma nth_le_tail (l : list α) (i) (h : i < l.tail.length)
  (h' : i + 1 < l.length := by simpa [←lt_sub_iff_right] using h) :
  l.tail.nth_le i h = l.nth_le (i + 1) h' :=
begin
  cases l,
  { cases h, },
  { simpa }
end
```
Renaming lemma to follow naming convention (like `nth_le_take`, `nth_le_drop`: named as if written without dot notation)
Provide in-bounds proof via opt_param, to help with rewrites and issues of syntactic equality, because `rw` or `simp` can make ugly proof terms
Golf the in-bounds proof and the nonempty list case
Induction isn't necessary, cases is sufficient.

Reply:
Thanks! The opt_param proof trick is nice, I should use that.

Reply:
I learned it from Anne =)